packages/core/src/server.ts

```diff
@@ -639,6 +669,20 @@ export class Controller extends API {
         await Promise.all([this.storage.delete(session), this.storage.save(auth)]);
 
         this.log("account.revokeSession", { revokedSession: { id, device: session.device } });
+
+        if (revokedForFailedAttempts && account.settings.failedLoginAttemptNotifications) {
+            try {
+                const location = this._buildLocationAndDeviceString(session.info.lastLocation, session.info.device);
+
+                // Send invite link to invitees email address
+                await this.messenger.send(account.email, new FailedLoginAttemptMessage({ location }));
+
+                // Remove trusted device, if it was (after email as this can throw if the device was already removed)
+                if (session.info.device) {
+                    await this.removeTrustedDevice(session.info.device.id);
+                }
+            } catch (e) {}
+        }
```

Comment:
Again, I think this is conflating failed login attempts with failed unlock attempts. I know a lot of people don't really understand the difference but it's important for us to make a clear distinction here. I would feel better if we'd do this as part of another PR.